onnxscript/rewriter/ort_fusions/sdpa.py

```diff
@@ -11,6 +11,7 @@
 class SDPA(pattern.RewriteRuleClassBase):
     def __init__(self, name: str, *, use_mask: bool, pre_scale: bool, use_mul: bool):
         super().__init__(name=name)
+        self.as_function = True
```

Comment:
I wonder if this can be controlled by the pattern engine instead of the rule itself.

Reply:
How? I assume you mean the engine should figure it out automatically, but that's hard. It could choose to extract the replaced set of nodes as a function whenever it is legal, but that could cause the model size to increase unnecessarily in some cases (at least temporarily). I guess it may be doable if we don't mind this performance overhead. We *need* it only for speculative fusions that introduce a temporary intermediate op (like SDPA above) that is expected to be subsequently removed by another fusion (like MHA or GQA) ... and if it is not, we would like to undo the original speculative fusion. Anyway: this is an interesting extension to consider later on.

Reply:
SG, thanks! Just a thought to explore